draft-ietf-jsonpath-base.md

```diff
@@ -296,10 +296,22 @@ stable JSON query language.
 ## Overview of JSONPath Expressions {#overview}
 
 JSONPath expressions are applied to a JSON value, the *argument*.
-Within the JSONPath expression, the abstract name `$` is used to refer
-to the *root node* of the argument, i.e., to the argument as a whole.
 
-JSONPath expressions use the *bracket notation*, for example:
+A JSONPath expression consists of an identifier followed by a series
+of zero or more segments each of which contains one or more selectors.
+
+### Identifiers
+
+The _root node identifier_ `$` refers to the *root node* of the argument,
+i.e., to the argument as a whole.
+Every JSONPath expression begins with the root node identifier.
+
+The _current node identifier_ `@` refers to the *current node* in the context
+of a filter expression (described later).
```

Comment:
I think this really means "in the context of the evaluation of a filter expression"?

Reply:
(Somehow off-topic here) I would like to avoid collision of traditional term "current node" in this context with its use in 3.3 Semantics
```
A segment performs its function on each of the nodes in its input
   nodelist, during such a function execution, such a node is referred
   to as the "current node",
```
There I propose to exchange  "current node" by "context node", which is used by XPath in a similar context.

Reply:
> (Somehow off-topic here) I would like to avoid collision of traditional term "current node" in this context with its use in 3.3 Semantics
> 
> ```
> A segment performs its function on each of the nodes in its input
>    nodelist, during such a function execution, such a node is referred
>    to as the "current node",
> ```
> 
> There I propose to exchange "current node" by "context node", which is used by XPath in a similar context.

@goessner Well spotted. Please can we address this separately to this PR? I suggest starting with an issue to hash out a consensus on the terminology.

Reply:
> (Somehow off-topic here) I would like to avoid collision of traditional term "current node" in this context with its use in 3.3 Semantics
> 
> ```
> A segment performs its function on each of the nodes in its input
>    nodelist, during such a function execution, such a node is referred
>    to as the "current node",
> ```
> 
> There I propose to exchange "current node" by "context node", which is used by XPath in a similar context.

@goessner Well spotted. Please can we address this separately to this PR? I suggest starting with an issue to hash out a consensus on the terminology.

> (Somehow off-topic here) I would like to avoid collision of traditional term "current node" in this context with its use in 3.3 Semantics
> 
> ```
> A segment performs its function on each of the nodes in its input
>    nodelist, during such a function execution, such a node is referred
>    to as the "current node",
> ```
> 
> There I propose to exchange "current node" by "context node", which is used by XPath in a similar context.

This discussed was moved to https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/285.

Reply:
> I think this really means "in the context of the evaluation of a filter expression"?

Adopted.